### PR TITLE
Deprecate the files in base/std_cxx11 -- we no longer need them.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/base/std_cxx11/array.h [hunk 1]
```diff
@@ -16,6 +16,7 @@
 #ifndef dealii_std_cxx11_array_h
 #define dealii_std_cxx11_array_h
 
+#warning "This file is deprecated. Simply use the corresponding C++11 header <array>."
```
Comment (drwells) [inline include/deal.II/base/std_cxx11/array.h]: I think that for MSVC compatibility we need to use `DEAL_II_WARNING` which is in `config.h`.

File: include/deal.II/base/std_cxx11/bind.h [hunk 2]
```diff
@@ -17,6 +17,8 @@
 #define dealii_std_cxx11_bind_h
 
 
+#warning "This file is deprecated. Simply use the corresponding C++11 header <functional>."
```
Comment (drwells) [inline include/deal.II/base/std_cxx11/bind.h]: etc.

**Comment by bangerth** [comment]
Ah, I had forgotten about `DEAL_II_WARNING`. Thanks for reminding me!

The patch is updated as requested now!

**Pull request merged**